src/go/k8s/pkg/resources/configmap.go

```diff
@@ -266,34 +333,39 @@ func (r *ConfigMapResource) createConfiguration(
 		if secretKeyStr == "" {
 			return nil, fmt.Errorf("secret name %s, ns %s: %w", secretName.Name, secretName.Namespace, errCloudStorageSecretKeyCannotBeEmpty)
 		}
-		r.prepareCloudStorage(cr, secretKeyStr)
+		r.prepareCloudStorage(cfg, secretKeyStr)
 	}
 
-	for _, user := range r.pandaCluster.Spec.Superusers {
-		cr.Superusers = append(cr.Superusers, user.Username)
+	if len(r.pandaCluster.Spec.Superusers) > 0 {
+		superusers, err := getSuperusersFromConfig(cfg)
+		if err != nil {
+			return nil, err
+		}
+		for _, user := range r.pandaCluster.Spec.Superusers {
+			superusers = append(superusers, user.Username)
+		}
```

Comment:
So there are 3 places where superusers are computed. The CR is one of them, but when you enable pandaproxy or schema registry, their users are also added to the list of superusers. So the code you see here is the equivalent of an `append` on an unstructured property.
It's true that in this particular spot the slice is supposed to be empty, but doing an append makes the code more robust to refactorings imo.

Btw, I think it's a good idea to put an `append` method for string slices in the global property object, so that I can avoid the boilerplate and also test it...

Reply:
Thank you for explanation